src/daft-connect/src/spark_analyzer.rs

```diff
@@ -832,179 +832,15 @@ impl SparkAnalyzer<'_> {
-                                    left_on.push(unresolved_col(name.clone()));
-                                    right_expr_join_side = Some(JoinSide::Left);
-                                } else if right_schema.has_field(name) {
-                                    right_on.push(unresolved_col(name.clone()));
-                                    right_expr_join_side = Some(JoinSide::Right);
-                                } else {
-                                    internal_err!("column not found");
-                                }
-                            }
-                        }
-                    }
-
-                    // make sure the join condition is valid
-                    // any non-column expression should also be handled here.
-                    // ex: col(z) = 1
-                    // if the column (z) is on the left side, then the right side should be the literal and vice versa
-                    // ensuring that there is an expression on both sides
-                    match (left_expr_join_side, right_expr_join_side) {
-                        (None, Some(JoinSide::Right)) => left_on.push(left.clone()),
-                        (Some(JoinSide::Left), None) => right_on.push(right.clone()),
-                        (Some(JoinSide::Right), None) => left_on.push(right.clone()),
-                        (None, Some(JoinSide::Left)) => right_on.push(left.clone()),
-                        (Some(JoinSide::Left), Some(JoinSide::Right))
-                        | (Some(JoinSide::Right), Some(JoinSide::Left)) => {}
-                        _ => {
-                            internal_err!("both sides of join condition are on the same side");
-                        }
-                    }
-                }
-                Expr::BinaryOp {
-                    op: Operator::And,
-                    left,
-                    right,
-                } => {
-                    process_join_on(left.clone(), left_on, right_on, left_plan, right_plan)?;
-                    process_join_on(right.clone(), left_on, right_on, left_plan, right_plan)?;
-                }
-                _ => {
-                    not_yet_implemented!("join condition");
-                }
-            }
-            Ok(())
-        }
-
-        let mut left_on = vec![];
-        let mut right_on = vec![];
-        if let Some(join_cond) = join_condition {
-            let join_cond = analyze_expr(&join_cond)?;
-            process_join_on(
-                join_cond,
-                &mut left_on,
-                &mut right_on,
-                &left_plan,
-                &right_plan,
-            )?;
-        } else {
-            let using_columns: Vec<ExprRef> =
-                using_columns.into_iter().map(unresolved_col).collect();
-            left_on.clone_from(&using_columns);
-            right_on.clone_from(&using_columns);
-        }
+        let on = join_condition.as_ref().map(analyze_expr).transpose()?;
```

Comment:
🔥

Reply:
Haha sorry for all that hard work that's now deleted though